### PR TITLE
Rotary Encoder Work

### LINKED ISSUE #32: Interface with Potentiometers
We will need to interface with the potentiometers in order to read:

- [x] Button click
- [x] Clockwise knob spins
- [x] Counter clockwise knob spins
- [x] Be able to determine which knobs each event is occurring for

### PR DESCRIPTION
**Pull request by mjschuetze102**
# Description
## Problem
Previously there was no interface for working with the rotary encoders to receive the values they were producing.

## Solution
Add in an interface to work with the rotary encoders, both with button clicks and spin.

Fixes #32

### CODE REVIEW

File: Brixcer_Box/platformio.ini [hunk 1]
```diff
@@ -12,3 +12,13 @@
...
+
+; [env:disco_l475vg_iot01a]
+; platform = atmelavr
+; board = uno
+; framework = arduino
+
+; [env:teensy31]
+; platform = teensy
+; board = teensy31
+; framework = arduino
```
Comment (Michael-Longley) [inline Brixcer_Box/platformio.ini]: Do we want to commit this?

File: Brixcer_Box/src/KnobControl.cpp [hunk 2]
```diff
@@ -9,14 +9,37 @@ KnobControl::KnobControl(int knobPinA, int knobPinB, int buttonPin) {
...
 }
 
-int* KnobControl::readKnobValues() {
-    int knobValues[] = {digitalRead(knobPinA), digitalRead(knobPinB)};
-    return knobValues;
+int KnobControl::readKnobValues() {
+    int knobValues[2] = {digitalRead(knobPinA), digitalRead(knobPinB)};
+
+    // Check if this knob was the activated one
+    int temp = 0;
```
Comment (Michael-Longley) [inline Brixcer_Box/src/KnobControl.cpp]: Can we rename this variable?

File: Brixcer_Box/src/main.cpp [hunk 3]
```diff
@@ -30,13 +30,18 @@
 #define Button1 D16
 #define LedChain D10
 
+
```
Comment (Michael-Longley) [inline Brixcer_Box/src/main.cpp:33]: Is this a sniffing thing? Why was this added?

File: Brixcer_Box/src/main.cpp [hunk 4]
```diff
@@ -49,20 +54,52 @@ void setup() {
...
 void loop() {
 
 }
 
-void sendData(int control, int action, int value) {
+int countDigits(int num) {
+    int digits = 0;
+
+    while (num) {
```
Comment (Michael-Longley) [inline Brixcer_Box/src/main.cpp:78]: Will this run forever? Could it get to a point where it doesn't get to 0?

File: Brixcer_Box/src/main.cpp (hunk 3 above)
Comment (mjschuetze102) [reply Brixcer_Box/src/main.cpp:33]: To separate out button definitions from other globals

File: Brixcer_Box/src/main.cpp (hunk 4 above)
Comment (mjschuetze102) [reply Brixcer_Box/src/main.cpp:78]: No, because it is defined as an int (i.e. truncates to whole number)

File: Brixcer_Box/platformio.ini (hunk 1 above)
Comment (mjschuetze102) [reply Brixcer_Box/platformio.ini]: Doesn't really matter. I removed it to see if GH will ignore the merge conflict (since you added the UNO to the list in one of your PRs.

Comment (mjschuetze102) [reply Brixcer_Box/platformio.ini]: It did not... I fixed the merge anyway

File: Brixcer_Box/src/main.cpp (hunk 3 above)
Comment (Michael-Longley) [reply Brixcer_Box/src/main.cpp:33]: okay sounds good. we should set up some sniffs. Not sure what they use in C++ though.

**Review by Michael-Longley — APPROVED** [review]
🚢 it
